Edge vsphere

7flying opened the pull request:
This adds the `edge-vsphere` image type for x86_64 and
aarch64. It is based on the `edge-raw-image` but produces
a .vmdk image in order to upload the artifact to vSphere.

This image has Ignition support.

This has been moved from [osbuild/osbuild-composer#3539](https://github.com/osbuild/osbuild-composer/pull/3539)

This pull request includes:

- [x] adequate testing for the new functionality or fixed issue (I think so?)
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

<!--
Thanks for proposing a change to osbuild-composer!

Please don't remove the above check list. These are things that each pull
request must have before it is merged. It helps maintainers to not forget
anything.

If the reason for ticking any of the boxes is ambiguous, please add a short
note explaining why.

In addition, if this pull request fixes a downstream issue, please refer to
test/README.md and add these additional items:

- [ ] 1st commit of any `rhbz#` related PR contains bug reproducer; CI reports FAIL or
- [ ] PR contains automated tests for new functionality and
- [ ] QE has approved reproducer/new tests and
- [ ] Subsequent commits provide bug fixes without modifying the reproducer; CI reports PASS and
- [ ] QE approves this PR; RHBZ status is set to `MODIFIED + Verified=Tested`

Information regarding our GitLab pipeline (Schutzbot):

CI will not be ran automatically if WIP label is applied or the PR is in DRAFT state, instead only
a link will be provided to the pipeline which can then be triggered manually if desired. To run the
CI automatically either switch the PR to ready or apply WIP+test label.

Outside contributors need manual approval from one of the osbuild-composer maintainers.

Schutzbot will only be triggered if all Tests jobs in GitHub workflow succeed.
-->

7flying commented:
From the latest review over the `osbuild-composer` repo:

Made changes on VMDK so that the `NewVMDK` function also takes a `imgOstreePipeline *RawOSTreeImage`, and changed the VMDK struct accordingly to accommodate that (it has a `Pipeline` instead of a `RawImage`) + cleanups related to that.

On `ostree_raw` we explicitly fail if a compression type is specified with the VMDK format.

7flying commented:
Simply changed the commit message names since there is no longer an `internal` folder :smiling_face_with_tear:

achilleas-k reviewed (changes requested):
LGTM.  Thanks for bringing this over.

I'm marking this as "Request Changes" for now because I'd like to block it until #26 is merged to bring in build tests for all the image types.
Keep in mind that #26 removes the static manifests from `test/data/manifests/`, so you can drop them from this PR too to avoid conflicts.

7flying commented:
Rebased on main and removed the manifest files from the PR.

achilleas-k commented:
The new image type depends on pulling an ostree commit, which I haven't set up yet (working on it now).  Can you add the new image type to the skip list here, please?
https://github.com/osbuild/images/blob/e61dd46dc12ae17910743e4432a4ab0ce2f0dc00/test/cases/generate-build-config#L24-L32

7flying commented:
> The new image type depends on pulling an ostree commit, which I haven't set up yet (working on it now). Can you add the new image type to the skip list here, please?
> 
> https://github.com/osbuild/images/blob/e61dd46dc12ae17910743e4432a4ab0ce2f0dc00/test/cases/generate-build-config#L24-L32

Rebased on main, added the  `edge-vsphere` image to that list and also added a missing `,` after `edge-ami` on the list above.

7flying commented:
it looks like the `edge-vsphere` tests aren't being skipped yet: https://gitlab.com/redhat/services/products/image-builder/ci/images/-/jobs/4686776024#L6252

achilleas-k commented:
Yes, the manifest generation needs to happen for all image types before it decides which ones to skip, so there needs to be a config for it.  Sorry, forgot to mention it.  I'll add it now.
